dvc/remote/gdrive.py

```diff
@@ -41,10 +41,43 @@ def __init__(self, path):
 def gdrive_retry(func):
     retry_re = re.compile(r"HttpError (403|500|502|503|504)")
 
+    def retry_error(exc):
+        import json
+
+        if RemoteGDrive.corpora != "drive" or not exc.args:
+            return True
+        http_error = exc.args[0]
+        content = json.loads(http_error.content)
+        if content and "error" not in content:
+            return True
+        err = content["error"]
+        if (
+            err["code"] == 403
+            and len(err["errors"])
+            and err["errors"][0].get("location", "") == "file.permissions"
+        ):
+            raise DvcException(
```

Comment:
we should catch the error in GC implementation method and convert it there, not in retry logic

Reply:
changed, thanks